tools/build_defs/framework.bzl

```diff
@@ -404,23 +429,29 @@ def _define_inputs(attrs):
     linking_infos_all = []
 
     bazel_headers = []
+    bazel_system_includes = []
     bazel_libs = []
 
     # This framework function-built libraries: copy result directories under
     # $EXT_BUILD_DEPS/lib-name
     ext_build_dirs = []
+    ext_build_dirs_set = {}
 
     for dep in attrs.deps:
-        provider = dep[OutputGroupInfo]
-        ext_built = provider and hasattr(provider, "gen_dir")
+        provider = dep[ExternallyBuiltTransitive]
 
         linking_infos_all += [dep[CcLinkingInfo]]
         compilation_infos_all += [dep[CcCompilationInfo]]
 
-        if ext_built:
-            ext_build_dirs += provider.gen_dir.to_list()
+        if provider:
+            for artifact in provider.artifacts:
+                if not ext_build_dirs_set.get(artifact.gen_dir):
+                    ext_build_dirs_set[artifact.gen_dir] = 1
```

Comment:
how big do you expect this set to get? is it worth it for the faster look up vs just checking if the item is in the list?